configure script cannot run twice in mips cross environment

Here is the output: https://github.com/abcfy2/qBittorrent-Enhanced-Edition/actions/runs/5119758583/jobs/9215933628

```
zlib-ng version 2.1.1-beta2
Using mips-linux-musl-ar
Using mips-linux-musl-ranlib
Checking for compiler... mips-linux-musl-gcc
Building static library libz.a version 1.2.13.zlib-ng with mips-linux-musl-gcc.
Checking for off64_t... Yes.
Checking for fseeko... Yes.
Checking for posix_memalign... Yes.
Checking for aligned_alloc... Yes.
Checking for strerror... Yes.
Checking for getauxval() or elf_aux_info() in sys/auxv.h... Yes.
Checking for unistd.h... Yes.
Checking for ptrdiff_t... Yes.
Checking for ANSI C compliant compiler...  Yes.
Checking for -fno-semantic-interposition... Yes.
Checking for -fno-lto... Yes.
Checking for attribute(visibility(hidden)) support... Yes.
Checking for attribute(visibility(internal)) support... Yes.
Checking for attribute(aligned) ... Yes.
Checking for _Thread_local ... Yes.
Checking for __builtin_ctz ... Yes.
Checking for __builtin_ctzll ... Yes.
ARCH: mips
Using arch directory: arch/generic
Architecture-specific static object files:
Architecture-specific shared object files:
grep: /usr/src/zlib-ng-2.1.1-beta2/arch/generic/*.c: No such file or directory
grep: /usr/src/zlib-ng-2.1.1-beta2/arch/generic/*.c: No such file or directory
Using cross-compile emulator: qemu-mips -L /usr/mips-linux-musl/
make: *** No rule to make target '/usr/src/zlib-ng-2.1.1-beta2/arch/generic/*.c', needed by 'example.o'.  Stop.
Error: Process completed with exit code 2.
```

And I've tried `make distclean`, but there's an error:

```sh
# make distclean
make[1]: Entering directory '/usr/src/zlib-ng-2.1.1-beta2/arch/generic'
rm -f *.o *.lo *~ \
rm -rf objs
rm -f *.gcda *.gcno *.gcov
make[1]: Leaving directory '/usr/src/zlib-ng-2.1.1-beta2/arch/generic'
make[1]: Entering directory '/usr/src/zlib-ng-2.1.1-beta2/test'
Makefile:22: *** recipe commences before first target.  Stop.
make[1]: Leaving directory '/usr/src/zlib-ng-2.1.1-beta2/test'
make: *** [Makefile:359: clean] Error 2
```

There might be something wrong in `configure` script. Please help.

Thanks.

When the first time run in a clean project directory, it works:

```sh
# CHOST="${CROSS_HOST}" ./configure --prefix="${CROSS_PREFIX}" --static --zlib-compat
Using mips64-linux-musl-ar
Using mips64-linux-musl-ranlib
Checking for compiler... mips64-linux-musl-gcc
Building static library libz.a version 1.2.13.zlib-ng with mips64-linux-musl-gcc.
Checking for off64_t... Yes.
Checking for fseeko... Yes.
Checking for posix_memalign... Yes.
Checking for aligned_alloc... Yes.
Checking for strerror... Yes.
Checking for getauxval() or elf_aux_info() in sys/auxv.h... Yes.
Checking for unistd.h... Yes.
Checking for ptrdiff_t... Yes.
Checking for ANSI C compliant compiler...  Yes.
Checking for -fno-semantic-interposition... Yes.
Checking for -fno-lto... Yes.
Checking for attribute(visibility(hidden)) support... Yes.
Checking for attribute(visibility(internal)) support... Yes.
Checking for attribute(aligned) ... Yes.
Checking for _Thread_local ... Yes.
Checking for __builtin_ctz ... Yes.
Checking for __builtin_ctzll ... Yes.
ARCH: mips64
Using arch directory: arch/generic
Architecture-specific static object files:
Architecture-specific shared object files:
grep: /usr/src/zlib-ng-2.1.1-beta2/arch/generic/*.c: No such file or directory
grep: /usr/src/zlib-ng-2.1.1-beta2/arch/generic/*.c: No such file or directory
Using cross-compile emulator: qemu-mips64 -L /usr/mips64-linux-musl/

# make
mips64-linux-musl-gcc -O2  -std=c11 -Wall -DNDEBUG -D_LARGEFILE64_SOURCE=1 -DHAVE_POSIX_MEMALIGN -DHAVE_ALIGNED_ALLOC -DHAVE_SYS_AUXV_H -DZLIB_COMPAT -DWITH_GZFILEOP -DHAVE_VISIBILITY_HIDDEN -DHAVE_VISIBILITY_INTERNAL -DHAVE_ATTRIBUTE_ALIGNED -DHAVE_THREAD_LOCAL -DHAVE_BUILTIN_CTZ -DHAVE_BUILTIN_CTZLL -DWITH_GZFILEOP -I/usr/src/zlib-ng-2.1.1-beta2 -c -o example.o /usr/src/zlib-ng-2.1.1-beta2/test/example.c
...
```

But when I run `configure` twice, then `make` will fail:

```sh
# CHOST="${CROSS_HOST}" ./configure --prefix="${CROSS_PREFIX}" --static --zlib-compat
Using mips64-linux-musl-ar
Using mips64-linux-musl-ranlib
Checking for compiler... mips64-linux-musl-gcc
Building static library libz.a version 1.2.13.zlib-ng with mips64-linux-musl-gcc.
Checking for off64_t... Yes.
Checking for fseeko... Yes.
Checking for posix_memalign... Yes.
Checking for aligned_alloc... Yes.
Checking for strerror... Yes.
Checking for getauxval() or elf_aux_info() in sys/auxv.h... Yes.
Checking for unistd.h... Yes.
Checking for ptrdiff_t... Yes.
Checking for ANSI C compliant compiler...  Yes.
Checking for -fno-semantic-interposition... Yes.
Checking for -fno-lto... Yes.
Checking for attribute(visibility(hidden)) support... Yes.
Checking for attribute(visibility(internal)) support... Yes.
Checking for attribute(aligned) ... Yes.
Checking for _Thread_local ... Yes.
Checking for __builtin_ctz ... Yes.
Checking for __builtin_ctzll ... Yes.
ARCH: mips64
Using arch directory: arch/generic
Architecture-specific static object files:
Architecture-specific shared object files:
grep: /usr/src/zlib-ng-2.1.1-beta2/arch/generic/*.c: No such file or directory
grep: /usr/src/zlib-ng-2.1.1-beta2/arch/generic/*.c: No such file or directory
Using cross-compile emulator: qemu-mips64 -L /usr/mips64-linux-musl/

# make
make: *** No rule to make target '/usr/src/zlib-ng-2.1.1-beta2/arch/generic/*.c', needed by 'example.o'.  Stop.

# make distclean
make[1]: Entering directory '/usr/src/zlib-ng-2.1.1-beta2/arch/generic'
rm -f *.o *.lo *~ \
rm -rf objs
rm -f *.gcda *.gcno *.gcov
make[1]: Leaving directory '/usr/src/zlib-ng-2.1.1-beta2/arch/generic'
make[1]: Entering directory '/usr/src/zlib-ng-2.1.1-beta2/test'
Makefile:22: *** recipe commences before first target.  Stop.
make[1]: Leaving directory '/usr/src/zlib-ng-2.1.1-beta2/test'
make: *** [Makefile:359: clean] Error 2
```

--- a/configure
+++ b/configure
@@ -2148,8 +2148,12 @@ sed < $SRCDIR/Makefile.in "
 /^ARCH_SHARED_OBJS *=/s#=.*#=$ARCH_SHARED_OBJS#
 " > Makefile
 
-# Append header files dependences.
+# Append header files dependencies.
 for file in $SRCDIR/*.c $SRCDIR/test/*.c $SRCDIR/test/fuzz/*.c $SRCDIR/$ARCHDIR/*.c $SRCDIR/tools/*.c; do
+    if test ! -f $file; then
+        continue
+    fi
+
     short_name=$(echo $file | sed -e "s#$SRCDIR/##g")
     incs=$(grep -h include $file | sed -n 's/# *\include *"\(.*\.h\)".*/\1/p' | sort | uniq)
     includes=$(for i in $incs; do
@@ -2222,8 +2226,12 @@ sed < $SRCDIR/$ARCHDIR/Makefile.in "
 /^PPCFLAGS *=/s#=.*#=$vmxflag#
 " > $ARCHDIR/Makefile
 
-# Append header files dependences.
+# Append header files dependencies.
 for file in $SRCDIR/$ARCHDIR/*.c; do
+    if test ! -f $file; then
+        continue
+    fi
+
     incs=$(grep -h include $file | sed -n 's/# *\include *"\(.*\.h\)".*/\1/p' | sort | uniq)
     includes=$(for i in $incs; do
                    # Check that the include file exists in the current dir,

